library/src/validations/ulid/ulid.ts

```diff
@@ -0,0 +1,28 @@
+import type { PipeResult } from '../../types';
+
+/**
+ * Creates a validation functions that validates a ULID.
+ *
+ * @link https://github.com/ulid/spec
+ *
+ * @param error The error message.
+ *
+ * @returns A validation function.
+ */
+export function ulid<TInput extends string>(error?: string) {
+  return (input: TInput): PipeResult<TInput> => {
+    if (
+      input.length !== 26 ||
```

Comment:
Is this necessary? The regex already checks the length. I'll probably remove it. Let me know if there is an important reason to keep it.

Reply:
I actually added the length for the specific case of having a valid ULID _plus_ extra characters. The Regex could probably be updated instead to `^[0123456789ABCDEFGHJKMNPQRSTVWXYZ]{26}$` so that it's _only_ those 26 characters. Let me see if that works for the tests I've added

Reply:
Yeah, good call. Using the `^regex$` format works as expected. Thanks for the callout here